manager/metrics/collector.go

```diff
@@ -0,0 +1,101 @@
+package metrics
+
+import (
+	"context"
+
+	"strings"
+
+	metrics "github.com/docker/go-metrics"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/state/store"
+)
+
+var (
+	nodesMetric metrics.LabeledGauge
+)
+
+func init() {
+	ns := metrics.NewNamespace("swarmkit", "", nil)
+	nodesMetric = ns.NewLabeledGauge("nodes", "The number of nodes", metrics.Total, "state")
```

Comment:
By Prometheus's metric naming conventions (https://prometheus.io/docs/practices/naming/), only cumulative counters have a `_total` suffix, while gauges that are just current counts don't have one - so just set the unit to `""` here. You'll wanna get to:

```
# HELP swarmkit_nodes_total The number of nodes
# TYPE swarmkit_nodes_total gauge
swarmkit_nodes{state="disconnected"} 0
swarmkit_nodes{state="down"} 0
swarmkit_nodes{state="ready"} 2
swarmkit_nodes{state="unknown"} 0
```

Btw., since Prometheus 1.6, `promtool` now has a `check-metrics` command that lints your metrics for this kind of thing. Maybe this is helpful for the future:

```
go get github.com/prometheus/prometheus/cmd/promtool
curl -s http://localhost:9090/metrics | promtool check-metrics
```